lib/rules/prefer-const.js

```diff
@@ -240,11 +259,32 @@ module.exports = {
          * @returns {void}
          */
         function report(node) {
-            context.report({
-                node: node,
-                message: "'{{name}}' is never reassigned, use 'const' instead.",
-                data: node
-            });
+            var reportArgs = {
+                    node: node,
+                    message: "'{{name}}' is never reassigned, use 'const' instead.",
+                    data: node
+                },
+                isNormalVariableDeclaration = node.parent.parent.type === "VariableDeclaration" &&
+                    (node.parent.parent.parent.type === "ForInStatement" ||
+                        node.parent.parent.parent.type === "ForOfStatement" ||
+                        node.parent.parent.type === "ForStatement" ||
+                        node.parent.init),
+                isDestructuring = node.parent.parent.type === "ObjectPattern" &&
+                    node.parent.parent.parent.parent.type === "VariableDeclaration",
+                variableDeclarationParent = findUp(node, "VariableDeclaration");
+
+            if (isNormalVariableDeclaration || isDestructuring) {
+
+                reportArgs.fix = function(fixer) {
```

Comment:
I'm not sure this is proper.

Because this `report()` function is called for each identifier node, and so this function may be called multiple times for a variable declaration node. I guess it needs to check whether all identifiers of this declaration are not reassigned.